qir/qat/Apps/Qat/ProfileConfiguration.hpp

```diff
@@ -0,0 +1,305 @@
+    configuration_manager.updateParameter("use-static-qubit-array-allocation", true);
+    configuration_manager.updateParameter("use-static-qubit-allocation", true);
+    configuration_manager.updateParameter("use-static-result-allocation", true);
+    configuration_manager.updateParameter("disable-reference-counting", true);
+    configuration_manager.updateParameter("disable-alias-counting", true);
+    configuration_manager.updateParameter("disable-string-support", true);
+    configuration_manager.updateParameter("disable-record-output-support", false);
+
+    configuration_manager.updateParameter("unroll-loops", false);
+    configuration_manager.updateParameter("allow-partial", true);
+    configuration_manager.updateParameter("allow-peeling", true);
+    configuration_manager.updateParameter("allow-runtime", true);
+    configuration_manager.updateParameter("allow-upper-bound", true);
+    configuration_manager.updateParameter("allow-profile-based-peeling", true);
+    configuration_manager.updateParameter<uint64_t>("full-unroll-count", 1024);
+    configuration_manager.updateParameter<int32_t>("unroll-opt-level", 3);
+    configuration_manager.updateParameter("only-when-forced", false);
+    configuration_manager.updateParameter("forget-scev", false);
+    configuration_manager.updateParameter("always-inline", true);
+    configuration_manager.updateParameter<int32_t>("inlining-parameter", 2147483647);
+    configuration_manager.updateParameter("eliminate-constants", true);
+    configuration_manager.updateParameter("eliminate-dead-code", true);
+    configuration_manager.updateParameter("eliminate-memory", true);
+
+    configuration_manager.updateParameter("delete-dead-code", false);
+    configuration_manager.updateParameter("clone-functions", false);
+    configuration_manager.updateParameter("transform-execution-path-only", false);
+    configuration_manager.updateParameter<uint64_t>("max-recursion", 1);
+    configuration_manager.updateParameter("assume-no-except", false);
+    configuration_manager.updateParameter("reuse-qubits", true);
+    configuration_manager.updateParameter("reuse-results", false);
+
+    configuration_manager.updateParameter("lower-switch", true);
+    configuration_manager.updateParameter("should-eleminate-zext-i1", true);
+    configuration_manager.updateParameter("pull-records-back", false);
+
+    configuration_manager.updateParameter("annotate-qubit-use", true);
+    configuration_manager.updateParameter("annotate-result-use", true);
+    configuration_manager.updateParameter("annotate-max-qubit-index", true);
+    configuration_manager.updateParameter("annotate-max-result-index", true);
+    configuration_manager.updateParameter("reindex-qubits", true);
+    configuration_manager.updateParameter("replace-qubit-on-reset", false);
+    configuration_manager.updateParameter("inline-after-id-change", true);
+
+    configuration_manager.updateParameter("separate-circuits", true);
```

Comment:
I'm a little surprised that this experimental setting is On by default (looks like it has been for a while). What effect does this have? Is this ignored when grouping is overall disabled (as on line 39)?

Reply:
Yes, you are right. The default should be off. I believe it has had no effect since production is always ran with `--disable-grouping` which effectively put it off.